src/main/java/de/tum/in/www1/artemis/domain/Result.java

```diff
@@ -483,18 +485,29 @@ public void filterSensitiveInformation() {
      * @param isBeforeDueDate if feedbacks marked with visibility 'after due date' should also be removed.
      */
     public void filterSensitiveFeedbacks(boolean isBeforeDueDate) {
-        feedbacks.removeIf(Feedback::isInvisible);
-
-        if (isBeforeDueDate) {
-            feedbacks.removeIf(Feedback::isAfterDueDate);
-        }
+        var filteredFeedback = createFilteredFeedbacks(isBeforeDueDate);
+        setFeedbacks(filteredFeedback);
 
         // TODO: this is not good code!
         var testCaseFeedback = feedbacks.stream().filter(Feedback::isTestFeedback).toList();
         setTestCaseCount(testCaseFeedback.size());
         setPassedTestCaseCount((int) testCaseFeedback.stream().filter(feedback -> Boolean.TRUE.equals(feedback.isPositive())).count());
     }
 
+    /**
+     * Returns a new list that only contains feedback that should be passed to the student.
+     * Does not change the feedbacks attribute of this entity.
+     *
+     * @see ResultDTO
+     *
+     * @param isBeforeDueDate if feedbacks marked with visibility 'after due date' should also be removed.
+     * @return the new filtered list
+     */
+    public List<Feedback> createFilteredFeedbacks(boolean isBeforeDueDate) {
```

Comment:
I find the parameter name to be a bit confusing: It sounds like its purpose it so say something about some feedback state (is it before the due date?). Maybe `filterAfterDueDateFeedbacks`?

Reply:
I renamed it to `removeHiddenFeedback` since we usually call these tests "hidden test". `filterAfterDueDateFeedbacks` sounds too long imo.